src/lib.rs

```diff
@@ -85,29 +117,7 @@ pub use self::write_all::{write_all, WriteAll};
 ///
 /// This trait importantly means that the `read` method only works in the
 /// context of a future's task. The object may panic if used outside of a task.
-pub trait AsyncRead: io::Read {
-    /// Tests to see if this I/O object may be readable.
-    ///
-    /// This method returns an `Async<()>` indicating whether the object
-    /// **might** be readable. It is possible that even if this method returns
-    /// `Async::Ready` that a call to `read` would return a `WouldBlock` error.
-    ///
-    /// There is a default implementation for this function which always
-    /// indicates that an I/O object is readable, but objects which can
-    /// implement a finer grained version of this are recommended to do so.
-    ///
-    /// If this function returns `Async::NotReady` then the current future's
-    /// task is arranged to receive a notification when it might not return
-    /// `NotReady`.
-    ///
-    /// # Panics
-    ///
-    /// This method is likely to panic if called from outside the context of a
-    /// future's task.
-    fn poll_read(&mut self) -> Async<()> {
```

Comment:
Perhaps these dletions could be left to a future PR? I think the reorganization is prudent but I'd be interested to get some more discussion on this specifically.